reciperadar/workers/recipes.py

```diff
@@ -20,6 +22,15 @@ def index_recipe(recipe_id):
         db.session.close()
         return
 
+    # Display only the oldest-known recipe of record; redirect all others to it
+    earliest_crawl = CrawlURL.find_earliest_crawl(recipe.dst)
+    if earliest_crawl:
+        src_hash = hash_bytes(earliest_crawl.url).encode("utf-8")
+        earliest_id = Recipe.generate_id(src_hash)
+        if recipe.id != earliest_id:
```

Comment:
Todo: precalculate `earliest_id` by storing it directly on the `CrawlURL` model.

Currently we _could_ replace the ID-to-ID comparison with a URL-to-URL comparison -- but the URLs in these cases would likely share long prefixes (the protocol part and domain name) making for unnecessary comparison work.  Probably cheaper than hashing, certainly, but if we can precalculate the hashes then that'd be even more efficient.

Reply:
(done)